### ISSUE TITLE
[Suggestion] choose the linux distro by license

### ISSUE THREAD

**Issue by ghost** [distrochooser/distrochooser#202]
**Is your feature request related to a problem? Please describe.**
This request is related to the issue: "Free vs non-free licenses" in [Change the questions depending on if they are (linux) beginnner](https://github.com/distrochooser/distrochooser/issues/149), issue: "prefer mainstream distributions over the smaller ones" in [[Suggestion] Prefer mainstream distributions over the smaller ones](https://github.com/distrochooser/distrochooser/issues/193) and issue: "metrics for distrochooser" in [[Suggestion] metrics for distrochooser](https://github.com/distrochooser/distrochooser/issues/200)

**Describe the solution you'd like**

> Some questions are not suitable to ask to beginners. They should not be asked, because they might be confusing.
> Examples of difficult questions are:
> 32bit vs 64bit
> Free vs non-free licenses
> Ways to administrate/install software. Here the answer 'I want to avoid systemd' is especially hard for linux beginners.

This can be done whether the user is a beginner or not. *For example, If there are reactive questions and answers, that question may or may not appear as technical. Please, see this: [Add answer defined additional selection layer](https://github.com/distrochooser/distrochooser/pull/185). For example...*

> **Do you have knowledge in linux?(yes|no) - if yes, technical questions are asked.**
> 1. select license
> 2. select package manager
> 3. select desktop environment or window manager or cli

#### ui/ux img 1
![image](https://user-images.githubusercontent.com/111701513/209587055-5469a020-31f1-4f45-9255-0aa744ee49b3.png)

#### ui/ux img 2
![image](https://user-images.githubusercontent.com/111701513/209587133-a1c36ec4-2507-4690-96c0-1a1c0782a583.png)

**Note:** As we can see in this scenario, there is a reactive question and answer. That depending on what is asked and answered, generates a new question and answer as well.

**Describe alternatives you've considered**
1. It is worth remembering that if the user wants to ask a question about security, privacy or anonymity, a question about the license of the linux distro is common
2. Another interesting solution without reactive question and answer includes, the idea of a user type. For example, depending on whether the user says he has a "business", is either a "home-user", is either a "end-user", is either a "normal user" or an "academic"... the license will match.

#### 1.1 Image 1
![image](https://user-images.githubusercontent.com/111701513/209587258-d071e799-9e97-49b6-bd69-1e1f8763cdbc.png)

#### 1.2. Image 2
<img src="https://user-images.githubusercontent.com/111701513/194731540-db3d1113-ff57-4928-b9e3-894f01b54a51.png" width="600px"/>

**image-description** Please, see this... this image is illustrative and is licensed with the [choosealicense.com/licenses/](https://choosealicense.com/licenses/). Here we have the comparison of various licenses and the permission and rights of the license.

#### 3. Concept
> **Do you own any business?**
> 1. (permission) commercial use? (yes/no)
> 2. (permission) patente use?  (yes/no)
> 3. (permission) private use?  (yes/no)
> 4. (condition) Disclose source? (yes/no)
> 5. (condition) license and copyright notice? (yes/no)
> 6. (condition) network use is distribution? (yes/no)
> 7. (condition) same license? (yes/no)
> 8. (condition) state changes? (yes/no)
> 9. (condition) none condition? (yes/no)

> **Are you just a user?**
> 1. (permission) distribution use?  (yes/no)
> 2. (permission) modification use?  (yes/no)
> 3. (permission) private use?  (yes/no)
> 4. (condition) none condition? (yes/no)

> **Are you academic?**
> 1. (permission) distribution use?  (yes/no)
> 2. (permission) modification use?  (yes/no)
> 3. (permission) commercial use? (yes/no)
> 4. (permission) patente use?  (yes/no)
> 5. (condition) none condition? (yes/no) 

#### 3. but... why? add license in distrochooser?

> GitHub satistics (stars, forks, issues) is a good metric if you compare projects on GitHub only, if a Distro does not use a public GitHub repo this will require an replacement metric

In general, you can ask the user to ignore any distros other than the one most likely to have conditioned or selected by the user. The most popular or least popular distro should in theory be the distro most likely to be used by the user. So, if the user answers many questions he may have a greater chance of having 1 distro than many.*

So... I think this resource is important for quantitative analysis, I thought about this possibility: *Another possibility is to analyze the available amount of distro licenses, if there are 45 mit licenses, the most popular distros will be licensed in MIT (The opposite is valid in terms of quantitative analysis... If there are only 10 GPL licensed distros, then it ignores the 5 MIT and shows only the 10 in GPL... So... you can use this same thinking to check how many search results a given distro appears or how many search results in scientific articles the distro appears to analyze which distro is more or less popular as well).*

![image](https://user-images.githubusercontent.com/111701513/211209626-885be423-0cc0-44bf-92c2-fbb65c1238ac.png)

```markdown
// source code
pie title choose the linux distro by license
    "MIT" : 100
    "GPL" : 80
    "Unlicensed" : 15
```

*In other words it's a "reliable metric" or a partial solution to solve the problem of distro being more or less popular*

**Additional context**
- [Change the questions depending on if they are (linux) beginnner](https://github.com/distrochooser/distrochooser/issues/149)
- I would like to choose linux distro by license
- Maybe this can solve the mentioned problem of selecting or not the license as a basic or technical question
- I prefer to open an issue on the topic: "Free vs non-free licenses" here
- It can be a basic question, if you define the type of user, and with the type of user, define a type of question
- This can be resolved here [choosealicense.com/licenses/](https://choosealicense.com/licenses/)

**Comment by cmllr** [comment]
Hey, thank you for the suggestion! As with #185, I am working on adding a more fine-grained selection based on given answers. As a first implementation, I am also aiming to cover the license question on it. Additional implementation changes might change this, but I think it's a good start to cover this issue aswell.

Regarding choosealicense, I think it is better to just link the license list due to following reasons:

- Effort for maintenance: The list is curated an (at least I assume) up do date
- Scope: The scope of distrochooser is to help beginners, and some aspects of licenses might be overkill for beginners in Linux
- Time required for it: I don't think I will be able to cover maintenance for a licensing overview and the distribution mapping

Currently, the tagging looks like shown in the following image. The selection offers a textual decscription about what it is about and an additional description text which in this case refers to choosealicense.com. I want to do similar on other topics to offer the user more information sources in case the user demands that.

![image](https://user-images.githubusercontent.com/2878201/209553566-eece1fd3-8ff5-4c04-b3f3-430c171fec83.png)

Each tag will give some extra points for matching distributions, allowing to get more diversification into the results.

**Comment by cmllr** [comment]
I've deployed #185 to https://beta.distrochooser.de which adds a second level selection to the ideology question and the answer "I want to use free licenses as much as I can". I think this can serve as a basic implementation for this issue. Feedback is appreciated!

I will leave this issue open for discussion as the final mapping of these tags still need to be defined on the production instance.

**Comment by xahodo** [comment]
So, I would suggest having various types of licenses as opposed to specific licenses. For example "free" vs "non-free" and "strict stance on free software (link explaining what free software is)" or "lax stance on free software".

**Comment by cmllr** [comment]
> So, I would suggest having various types of licenses as opposed to specific licenses. For example "free" vs "non-free" and "strict stance on free software (link explaining what free software is)" or "lax stance on free software".

Valid argument. But I want to test this out how the user reacts. I am currently aiming to go for the licenses itself (also mostly as I can get these information automated), and if this does not result in the desired effects to make them more in way you suggested.

**Comment by cmllr** [comment]
hey @daskpunk, thanks for the link!

I don't understand this comment, unfortunately 

> As I understand distrochooser's metrics also work the same way, depending on how the user uses distrochooser - certain inputs and outputs are generated or made.
